src/cascadia/TerminalApp/Pane.h

```diff
@@ -19,155 +19,137 @@
 // - Mike Griese (zadjii-msft) 16-May-2019
 
 #pragma once
-#include <winrt/Microsoft.Terminal.TerminalControl.h>
 #include <winrt/TerminalApp.h>
 #include "../../cascadia/inc/cppwinrt_utils.h"
 
-enum class Borders : int
-{
-    None = 0x0,
-    Top = 0x1,
-    Bottom = 0x2,
-    Left = 0x4,
-    Right = 0x8
-};
-DEFINE_ENUM_FLAG_OPERATORS(Borders);
+class LeafPane;
+class ParentPane;
 
-class Pane : public std::enable_shared_from_this<Pane>
+class Pane
 {
 public:
-    Pane(const GUID& profile,
-         const winrt::Microsoft::Terminal::TerminalControl::TermControl& control,
-         const bool lastFocused = false);
-
-    std::shared_ptr<Pane> GetActivePane();
-    winrt::Microsoft::Terminal::TerminalControl::TermControl GetTerminalControl();
-    std::optional<GUID> GetFocusedProfile();
-
-    winrt::Windows::UI::Xaml::Controls::Grid GetRootElement();
-
-    bool WasLastFocused() const noexcept;
-    void UpdateVisuals();
-    void ClearActive();
-    void SetActive();
-
-    void UpdateSettings(const winrt::Microsoft::Terminal::Settings::TerminalSettings& settings,
-                        const GUID& profile);
-    void ResizeContent(const winrt::Windows::Foundation::Size& newSize);
-    void Relayout();
-    bool ResizePane(const winrt::TerminalApp::Direction& direction);
-    bool NavigateFocus(const winrt::TerminalApp::Direction& direction);
-
-    bool CanSplit(winrt::TerminalApp::SplitState splitType);
-    std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Split(winrt::TerminalApp::SplitState splitType,
-                                                                  const GUID& profile,
-                                                                  const winrt::Microsoft::Terminal::TerminalControl::TermControl& control);
-    float CalcSnappedDimension(const bool widthOrHeight, const float dimension) const;
+    virtual ~Pane() = default;
+    Pane(const Pane&) = delete;
+    Pane(Pane&&) = delete;
+    Pane& operator=(const Pane&) = delete;
+    Pane& operator=(Pane&&) = delete;
+
+    winrt::Windows::UI::Xaml::Controls::Grid GetRootElement() const;
+
+    // Method Description:
+    // - Searches for last focused pane in the pane tree and returns it. Returns
+    //   nullptr, if the pane is not last focused and doesn't have any child
+    //   that is.
+    // - This Pane's control might not currently be focused, if the tab itself is
+    //   not currently focused.
+    // Return Value:
+    // - nullptr if we're a leaf and unfocused, or no children were marked
+    //   `_lastActive`, else returns this

```

Comment:
Yea but having this on implementation would mean that all of this comments would be basically duplicated on both LeafPane and ParentPane. I don't remember seeing other (almost) pure virtual classes in this project though, so it might be that you have established different pattern.